### DISCUSSION TITLE
Conections hangs

### DISCUSSION THREAD

**Discussion by zeitler** [rancher/rke2#4607] [Q&A]
Hi all

Two different clusters having the same issue.
Both in debian 11 with kernel 5.10

Applications started to hang. After refresh the page sometimes it was ok
I've found some info about a bug on debian, but fixed after kernel 5.7

I wasn't aware that the issue was only on calico cni, or the fix only works on calico cni.
So, in one of the clusters, I've changed config file to start using cni. Same result

Changed mtu in to 1450, same result.

Found [this](https://slack-archive.rancher.com/t/2584353/afternoon-all-i-m-having-dns-issues-with-a-recently-installe)
and used `ethtool -K flannel.1 tx-checksum-ip-generic off`, same thing.

I've seen in tcpdump some checksum errors, that's why I've tried this fixes. But normally people say that they have some issues with coredns. 
I don't have that.

And I've installed a daemonset in the cluster, the one with canal cni. And when I curl the the endpoint from the same node there is no problems. From another cluster, it hangs but after a lot of curls. 
If I curl to the service it self, never hangs also.

This tests are for's curling the endpoint, several at a time.

I don't find a padron, and I'm out of ideas how to solve this.

Someone has some pointers? 
Thank you

**Comment by manuelbuil** [comment]
How are you installing rke2? If it is by config file, could you please share it?

**Reply by zeitler** [reply]
Hi

Thank you for your reply

I'm installing curling install.sh and, defining:
```
export INSTALL_RKE2_VERSION=v1.24.10+rke2r1
export INSTALL_RKE2_TYPE=server
sh install.sh
```

The config in one of the clusters:
```
tls-san:
  - <frontend IP>
  - <backend IP>
  - <Hostname>
node-label:
  - "node=master"
write-kubeconfig-mode: "0644"
debug: true
disable:
  - rke2-ingress-nginx
```

The other cluster, the same but with
`cni: calico` 

Installed ingress-nginx with:
``` 
helm upgrade --install ingress-nginx ingress-nginx \
--repo https://kubernetes.github.io/ingress-nginx \
--namespace ingress-nginx --create-namespace \
--version 4.5.2 \
--set controller.kind=DaemonSet --set controller.service.type=NodePort --set controller.hostPort.enabled=true --set controller.hostPort.ports.http=80 --set controller.hostPort.ports.https=443 \
```

I have the exact same configuration in another three clusters, but the OS is RHEL 8.6, with kernel 4.18.0 and it works very nicely

**Comment by manuelbuil** [comment]
Can you clarify: `when I curl the the endpoint from the same node there is no problems. From another cluster, it hangs but after a lot of curls.`

Does that mean two pods in the same cluster can communicate but two pods in different clusters can't communicate?

**Reply by zeitler** [reply]
My tests were from my computer `for i in {1..10000}; do curl https://myapp.somewhere.com/static; done`
Launched a pod in the cluster, and changing `https://myapp.somewhere.com` to `http://fe.namespace.svc.cluster.local:80` it never hangs.
It had hanged once. But it was 9 cicles in 1000 repetition, and only one for.

**Reply by manuelbuil** [reply]
Sorry, I don't understand. What is `myapp.somewhere.com/static` and what is `fe.namespace.svc.cluster.local`? Are those services exposed by your cluster? The curl to the first one works always and the curl to the second one works 9 out of 10000?

**Reply by zeitler** [reply]
I tend not to explain correctly...

`myapp.somewhere.com' is the url in the ingress deployed in the cluster.
`fe.namespace.svc.cluster.local` is the path for the service that connects the ingress and the application.

It doens't have a padron. Sometimes it hangs at first curl, sometimes in the 100th... 

And what you asked me for clarify: From anoter **node** and not **cluster** as written above.
I deployed a daemonset to debug if the problem was when the traffic was entering/exiting the cluster to the client or between nodes when the request was between ingress and aplication

In googling I found somewhere that the problem could be also due vmxnet3 vmware driver.
I was comparing the system with debian, and the other that works but is installed with RHEL.
Debian has version 1.5.0.0-k-NAPI and RHEL with 1.7.0.0-k-NAPI.
Are you aware of any issues with vmware driver?

**Reply by manuelbuil** [reply]
This is what I understood, please tell me if this is correct.

1 - Communication from pod1 to pod2 when both pods are in the same node works
2 - Communication from pod1 to pod2 when pods are in different nodes, does not always work, only sometimes
3 - Communication to service implemented by pod2 and accessed from the outside of the cluster always work

Is this correct?

**Reply by zeitler** [reply]
1- Correct
2 - Mostly works. I had only one fail
3 - Incorrect. It hangs without a padron

**Reply by manuelbuil** [reply]
How do you connect from pod1 to pod2? Are you using curl directly to the pod2 IP? Or are you using the service implemented by pod2?

**Reply by zeitler** [reply]
by service.

**Reply by manuelbuil** [reply]
How are you exposing the service to the outside of the cluster? Note that by default, services are of "ClusterIP" type, which means they are not accessible outside of the cluster

**Reply by zeitler** [reply]
Sorry taking your time.
As it seems, packet lost it's something internal in the network where these clusters are. An already known issue in an internal network, that never happen in kubernetes clusters.
Because most of our clusters are in rke, I thought changing for rke2 could be a cause for packet lost. 
A colleague told me about this situation that I was not aware.
After, I've made a connection through another network and packet loss stoped.

Even so, I'm having another issue. And again without clues on how to find why or solve it.

Communication between nodes it's quite slow.
If I have everything in the same node, the responses are quite fast.
Having pod1 in node1, pod2 in node2, increases at least for 1.5 seconds the request.

I have 3 different interfaces. Being the default the FE network. So rke2 uses that one. Even so, I've changed node-ip and node-external-ip without any results.
Googled and found that it could be dns query's with some issues. Logging rke2-coredns-rke2-coredns doesn't seem to have any issues.

Can you help me please on how to debug why this is happening?

**Reply by manuelbuil** [reply]
can you ping between nodes and check if you see that 1.5 seconds?

**Reply by zeitler** [reply]
From 0.215 to 0.350 ms.

**Reply by zeitler** [reply]
all nodes to the others

**Reply by manuelbuil** [reply]
Performance issues is always a bit complicated to debug. If I were you, I'd try to understand the path that the packet is following. Perhaps it is not using a direct route but being moved around without any purpose. Perhaps it is being natted multiple times accidentally

**Reply by zeitler** [reply]
Hi

The issue is due vxlan checksum bug
The solution in https://github.com/projectcalico/calico/issues/3145 fixed it

I had tried it before. But in the wrong interface.

Thanks for your help
